File: Chapter4/R/Example8.R
# Chapter 4 - Poisson Distribution
# Example 8: Call Center — Two Departments & Additive Property + Bayes
#   X ~ Poisson(4)  Technical Support
#   Y ~ Poisson(6)  Sales
#   X + Y ~ Poisson(10)  (additive property)

# Parameters
lambda_tech  <- 4    # Technical Support
lambda_sales <- 6    # Sales

cat("=== Call Center Analysis ===\n\n")

# i. P(Total = 8) using additive property
lambda_total   <- lambda_tech + lambda_sales   # = 10
prob_total_8   <- dpois(8, lambda_total)
cat(sprintf("i. P(Total = 8 calls) = %.4f\n\n", prob_total_8))

# ii. Conditional probability P(Tech=3 | Total=8)
# P(Tech=k, Sales=8-k) for k = 0,...,8
k_values    <- 0:8
probs_joint <- numeric(9)
for (k in k_values) {
  probs_joint[k + 1] <- dpois(k, lambda_tech) * dpois(8 - k, lambda_sales)
}

prob_tech_3_given_8 <- probs_joint[4] / sum(probs_joint)
cat(sprintf("ii. P(Tech=3 | Total=8) = %.4f\n\n", prob_tech_3_given_8))
cat(sprintf("    Verification: sum(joint probs) = %.4f\n",   sum(probs_joint)))
cat(sprintf("    (should equal P(Total=8) = %.4f)\n\n",      prob_total_8))

# iii. P(Total >= 10)
prob_at_least_10 <- 1 - ppois(9, lambda_total)
cat(sprintf("iii. P(Total >= 10 calls) = %.4f\n\n", prob_at_least_10))

# Visualization
par(mfrow = c(2, 2))

x_tech <- 0:15
plot(x_tech, dpois(x_tech, lambda_tech), type = 'h', lwd = 3,
     col  = 'blue', xlab = 'Number of Calls', ylab = 'Probability',
     main = sprintf('Technical Support\nPoisson(%d)', lambda_tech))
abline(v = lambda_tech, col = 'red', lty = 2, lwd = 2)

x_sales <- 0:20
plot(x_sales, dpois(x_sales, lambda_sales), type = 'h', lwd = 3,
     col  = 'darkgreen', xlab = 'Number of Calls', ylab = 'Probability',
     main = sprintf('Sales\nPoisson(%d)', lambda_sales))
abline(v = lambda_sales, col = 'red', lty = 2, lwd = 2)

x_total <- 0:25
plot(x_total, dpois(x_total, lambda_total), type = 'h', lwd = 3,
     col  = 'purple', xlab = 'Number of Calls', ylab = 'Probability',
     main = sprintf('Total Calls\nPoisson(%d)', lambda_total))
abline(v = lambda_total, col = 'red',    lty = 2, lwd = 2)
abline(v = 8,            col = 'orange', lty = 3, lwd = 2)
legend('topright', legend = c('Mean', 'Total=8'),
       col = c('red', 'orange'), lty = c(2, 3), lwd = 2, cex = 0.8)

# Conditional distribution P(Tech=k | Total=8)
conditional_probs <- probs_joint / sum(probs_joint)
barplot(conditional_probs, names.arg = k_values, col = 'coral',
        xlab = 'Technical Calls (k)', ylab = 'P(Tech=k | Total=8)',
        main = 'Conditional Distribution\nP(Tech=k | Total=8)')

par(mfrow = c(1, 1))

# Results table
cat("\n=== Conditional Probabilities Table ===\n")
cat("k\tP(Tech=k, Sales=8-k)\tP(Tech=k | Total=8)\n")
cat(strrep("-", 60), "\n")
for (k in k_values) {
  cat(sprintf("%d\t%.6f\t\t%.4f\n", k,
              probs_joint[k + 1], conditional_probs[k + 1]))
}

File: Chapter4/Python/Example8.py
# Chapter 4 - Poisson Distribution
# Example 8: Call Center — Two Departments & Additive Property + Bayes
#   X ~ Poisson(4)  Technical Support
#   Y ~ Poisson(6)  Sales
#   X + Y ~ Poisson(10)  (additive property)

import numpy as np
import matplotlib.pyplot as plt
from scipy.stats import poisson

# Parameters
lambda_tech  = 4    # Technical Support
lambda_sales = 6    # Sales

print("=== Call Center Analysis ===\n")

# i. P(Total = 8) using additive property
lambda_total = lambda_tech + lambda_sales   # = 10
prob_total_8 = poisson.pmf(8, lambda_total)
print(f"i. P(Total = 8 calls) = {prob_total_8:.4f}\n")

# ii. Conditional probability P(Tech=3 | Total=8)
k_values    = np.arange(0, 9)
probs_joint = np.array([
    poisson.pmf(k, lambda_tech) * poisson.pmf(8 - k, lambda_sales)
    for k in k_values
])

prob_tech_3_given_8 = probs_joint[3] / np.sum(probs_joint)
print(f"ii. P(Tech=3 | Total=8) = {prob_tech_3_given_8:.4f}\n")
print(f"    Verification: sum(joint probs) = {np.sum(probs_joint):.4f}")
print(f"    (should equal P(Total=8) = {prob_total_8:.4f})\n")

# iii. P(Total >= 10)
prob_at_least_10 = 1 - poisson.cdf(9, lambda_total)
print(f"iii. P(Total >= 10 calls) = {prob_at_least_10:.4f}\n")

# Visualization
fig, axes = plt.subplots(2, 2, figsize=(14, 10))

x_tech = np.arange(0, 16)
axes[0, 0].stem(x_tech, poisson.pmf(x_tech, lambda_tech),
                basefmt=' ', linefmt='blue', markerfmt='bo')
axes[0, 0].axvline(lambda_tech, color='red', linestyle='--', linewidth=2)
axes[0, 0].set_xlabel('Number of Calls')
axes[0, 0].set_ylabel('Probability')
axes[0, 0].set_title(f'Technical Support\nPoisson({lambda_tech})')
axes[0, 0].grid(alpha=0.3)

x_sales = np.arange(0, 21)
axes[0, 1].stem(x_sales, poisson.pmf(x_sales, lambda_sales),
                basefmt=' ', linefmt='darkgreen', markerfmt='go')
axes[0, 1].axvline(lambda_sales, color='red', linestyle='--', linewidth=2)
axes[0, 1].set_xlabel('Number of Calls')
axes[0, 1].set_ylabel('Probability')
axes[0, 1].set_title(f'Sales\nPoisson({lambda_sales})')
axes[0, 1].grid(alpha=0.3)

x_total = np.arange(0, 26)
axes[1, 0].stem(x_total, poisson.pmf(x_total, lambda_total),
                basefmt=' ', linefmt='purple', markerfmt='mo')
axes[1, 0].axvline(lambda_total, color='red',    linestyle='--', linewidth=2, label='Mean')
axes[1, 0].axvline(8,            color='orange', linestyle=':',  linewidth=2, label='Total=8')
axes[1, 0].set_xlabel('Number of Calls')
axes[1, 0].set_ylabel('Probability')
axes[1, 0].set_title(f'Total Calls\nPoisson({lambda_total})')
axes[1, 0].legend()
axes[1, 0].grid(alpha=0.3)

conditional_probs = probs_joint / np.sum(probs_joint)
axes[1, 1].bar(k_values, conditional_probs, color='coral', edgecolor='black', alpha=0.7)
axes[1, 1].axvline(3, color='blue', linestyle='--', linewidth=2)
axes[1, 1].set_xlabel('Technical Calls (k)')
axes[1, 1].set_ylabel('P(Tech=k | Total=8)')
axes[1, 1].set_title('Conditional Distribution\nP(Tech=k | Total=8)')
axes[1, 1].grid(alpha=0.3)

plt.tight_layout()
plt.show()

# Results table
print("\n=== Conditional Probabilities Table ===")
print("k\tP(Tech=k, Sales=8-k)\tP(Tech=k | Total=8)")
print("-" * 60)
for k in k_values:
    print(f"{k}\t{probs_joint[k]:.6f}\t\t{conditional_probs[k]:.4f}")
